lib/relay.js

```diff
@@ -8,12 +8,41 @@ const replace = require('./replace-vars');
 const tryJSONParse = require('./try-json-parse');
 const logger = require('./log');
 const version = require('./version');
+const stream = require('stream');
 
 module.exports = {
   request: requestHandler,
   response: responseHandler,
+  streamingResponse: streamResponseHandler,
 };
 
+const binaryRequest = [
+  'application/vnd.docker.image.rootfs.diff.tar.gzip',
+];
+
+const streams = new Map();
```

Comment:
Maybe there is a better way to bind the server response to the chunk handler. I do think the connection pooling gets in the way and I haven't found a way to remove event listeners from a socker i..e `off(streamResponseHandler)`

For now each response object of a binary streaming request gets a UUID which the streamResponseHandler uses to send the incoming chunks into the right response.

Reply:
I think this is fairly a correct approach to handle this async collection of binary chunks.
Consider adding some mechanism to limit the maximum memory consumption together with an ageing limit to handle the case where chunks streaming was stopped from some reason.

Reply:
I think the aging limit is a fair point and I'll add a TTL to the entries (using node-cache). 

In terms of memory consumption: We create a streamingID for each binary download - we would need a fair amount of downloads before we reach any kind of significant memory limit (the streaming objects are only PassThrough objects and should not consume a lot of memory). I think I would prefer to wait here and see if we run into memory issues before considering this.